### PR TITLE
Move more common classes into the only library that uses them

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Remove some unused constants from CommonConstants, remove some unnecessary client-side validation of user generated device/module symmetric keys

### CODE REVIEW

File: provisioning/service/src/Microsoft.Azure.Devices.Provisioning.Service.csproj [hunk 1]
```diff
@@ -46,9 +46,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <Compile Include="$(common)\service\CommonResources.cs" Link="Common\CommonResources.cs" />
     <Compile Include="$(common)\service\Resources.Designer.cs" Link="Common\Resources.Designer.cs" />
-    <Compile Include="$(common)\service\StringValidationHelper.cs" Link="Common\StringValidationHelper.cs" />
```
Comment (timtay-microsoft) [inline provisioning/service/src/Microsoft.Azure.Devices.Provisioning.Service.csproj:51]: Both these classes were unused in the provisioning service client

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**